src/app/test_executive/snapps_test.ml

```diff
@@ -31,40 +41,224 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
+        in
+        Signature_lib.Schnorr.Chunked.sign sk
+          (Random_oracle.Input.Chunked.field commitment)
+      in
+      (* we also need to update the other parties signatures, if there's a full commitment,
+         which relies on the fee payer hash
+      *)
+      let%bind other_parties_with_valid_signatures =
+        Malleable_error.List.mapi
+          parties_valid_fee_payer_signature.other_parties
+          ~f:(fun ndx { data; authorization } ->
+            (* 0th node has keypair for fee payer, so start at 1 *)
+            let node = List.nth_exn block_producer_nodes (ndx + 1) in
+            let%map sk = Util.priv_key_of_node node in
+            let authorization_with_valid_signature =
+              match authorization with
+              | Mina_base.Control.Signature _dummy ->
+                  let use_full_commitment = data.body.use_full_commitment in
+                  let signature =
+                    sign_for_other_party ~use_full_commitment sk
+                  in
+                  Mina_base.Control.Signature signature
+              | Proof _ | None_given ->
+                  authorization
+            in
+            { Mina_base.Party.data
+            ; authorization = authorization_with_valid_signature
+            })
+      in
+      return
+        { parties_with_other_parties_nonces with
+          other_parties = other_parties_with_valid_signatures
+        }
+    in
+    (* the generated parties0 has a fee payer, other parties with public
+       keys not found in the integration test ledger, so we replace those keys
+    *)
+    let%bind fee_payer_pk = Util.pub_key_of_node node in
+    let%bind other_parties_with_valid_keys =
+      Malleable_error.List.mapi parties0.other_parties
+        ~f:(fun ndx { data; authorization } ->
+          (* 0th node has keypair for fee payer, so start at 1 *)
+          let node = List.nth_exn block_producer_nodes (ndx + 1) in
+          let%map pk = Util.pub_key_of_node node in
+          let data = { data with body = { data.body with public_key = pk } } in
+          { Mina_base.Party.data; authorization })
+    in
+    let parties_valid_pks =
+      { parties0 with
+        fee_payer =
+          { parties0.fee_payer with
+            data =
+              { parties0.fee_payer.data with
+                body =
+                  { parties0.fee_payer.data.body with
+                    public_key = fee_payer_pk
+                  }
+              }
+          }
+      ; other_parties = other_parties_with_valid_keys
+      }
+    in
+    let%bind parties_valid =
+      mk_parties_with_signatures ~fee_payer_nonce:Unsigned.UInt32.zero
+        parties_valid_pks
+    in
+    (* choose payment sender, receiver that are not in the snapp other parties
+       because there's a check that user commands can't involve snapp accounts
+       TODO: that check will be removed; change sender to snapp fee payer
```

Comment:
The generator is probably generating some update for the fee payer party that is causing it to become a snapp account. I think substituting `Party.Update.noop` for fee payer update would fix this